### ISSUE TITLE
Tavern sometimes times out

### ISSUE THREAD

**Issue by Almad** [dracidoupe/graveyard#250]
Happened to @jimmeak and finally to me as well:

```
2021-05-06T11:33:08.463146+00:00 heroku[router]: at=error code=H12 desc="Request timeout" method=GET path="/putyka/?vypis=oblibene" host=nove.dracidoupe.cz request_id=b805ef6b-564d-4862-84dc-7308351d6f1b fwd="213.29.32.47" dyno=web.1 connect=0ms service=30000ms status=503 bytes=0 protocol=http
2021-05-06T11:33:08.938499+00:00 heroku[router]: at=info method=GET path="/favicon.ico" host=nove.dracidoupe.cz request_id=e819e81e-a8c4-45da-830d-4c279ed81ae6 fwd="213.29.32.47" dyno=web.1 connect=0ms service=80ms status=404 bytes=266 protocol=http
2021-05-06T11:33:08.937045+00:00 app[web.1]: 2021-05-06 13:33:08 [WARNING] Not Found: /favicon.ico
2021-05-06T11:33:08.937069+00:00 app[web.1]: 2021-05-06 13:33:08 [WARNING] Not Found: /favicon.ico
2021-05-06T11:33:08.937477+00:00 app[web.1]: 10.1.7.55 - - [06/May/2021:13:33:08 +0200] "GET /favicon.ico HTTP/1.1" 404 77 "http://nove.dracidoupe.cz/putyka/?vypis=oblibene" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:89.0) Gecko/20100101 Firefox/89.0"
2021-05-06T11:33:08.981794+00:00 app[web.1]: [2021-05-06 11:33:08 +0000] [4] [CRITICAL] WORKER TIMEOUT (pid:11)
2021-05-06T11:33:09.987822+00:00 app[web.1]: [2021-05-06 11:33:09 +0000] [14] [INFO] Booting worker with pid: 14
```

**Comment by Almad** [comment]
Related:

```
2021-05-06T11:33:13.130773Z 19143934 [Note] Aborted connection 19143934 to db: 'dracidoupe_cz' user: 'heroku_application' host: 'ec2-xxx.eu-west-1.compute.amazonaws.com' (Got an error reading communication packets)
```

**Comment by Almad** [comment]
Something apparently causes read IOPS spike:

<img width="1162" alt="image" src="https://user-images.githubusercontent.com/44565/117292551-a43d8f00-ae70-11eb-9ea9-72c416c87339.png">

**Comment by Almad** [comment]
Well, to take out:

* This happens at a DB level, not app level
* Can be therefore solved using caching, potentially
* I think the caching of total article count should help

For diagnostics though:

* Proper monitoring—at least slowlog and enhanced monitoring
* Part of production setup though, let me make a ticket on that

**Comment by Almad** [comment]
We are also on a 20G magnetic charged for iops, it's a question whether it wouldn't make sense to upgrade go ssd that's more expensive, but without iops charges / with throttles instead

**Comment by Almad** [comment]
Happened again after app cold start. My best guess is warm caches still.

**Comment by Almad** [comment]
Should be fixed through #308, let's see.
